core/editor/src/tools/select.rs

```diff
@@ -0,0 +1,40 @@
+use crate::events::Event;
+use crate::events::MouseKeys;
+use crate::tools::Tool;
+use document_core::Operation;
+
+#[derive(Default)]
+pub struct Select(Fsm);
+
+impl Tool for Select {
+	fn handle_input(&mut self, event: Event) -> Option<Operation> {
+		match event {
+			Event::MouseDown(state) => {
+				if !(state.mouse_keys & MouseKeys::LEFT).is_empty() {
```

Comment:
use `!state.mouse_keys.contains(MouseKays::LEFT)` instead.

Reply:
Also I think there can only ever be a single mouse key pressed per event, so that could even be `state.mouse_keys != MouseKeys::LEFT` with a comment stating what I just said.

Reply:
oh right